### DISCUSSION TITLE
ContainerUtils repository for other than openSUSE Leap Micro 5.5 systems.

### DISCUSSION THREAD

**Discussion by j-kihet** [uyuni-project/uyuni#9000] [Q&A] [answered]
ContainerUtils repository has packages for other operating systems than openSUSE Leap Micro 5.5.
Does this mean that migrating from a legacy installation to a container version is possible on other operating systems as well?
If its true, will the ContainerUtils also be updated for other operating systems going forward and new operating systems will be added such as Ubuntu 24.04?

**Comment by srbarrios** [comment]
For now, from the QE team perspective, even if we are building mgr-tools aka uyuni-tools for multiple operating systems, we will only cover openSUSE Leap Micro 5.5, as that's the only one described in the official Uyuni documentation.
From my personal opinion, I tried it in a regular Leap, but not in a Ubuntu yet, but I did not found any technical impediment, it would be good to see it in new versions of these distros, if it doesn't cause too much trouble.

**Comment by agraul** [comment]
Another aspect is that `uyuni-tools` is a collection of tools, not all of them are to be executed on the server. `mgrctl` can be used from any machine to interact with the Uyuni server container, similar to the Uyuni API client `spacecmd`.

**Comment by deneb-alpha** [comment, marked as answer]
hi @j-kihet 

> ContainerUtils repository has packages for other operating systems than openSUSE Leap Micro 5.5. Does this mean that migrating from a legacy installation to a container version is possible on other operating systems as well? If its true, will the ContainerUtils also be updated for other operating systems going forward and new operating systems will be added such as Ubuntu 24.04?

In the future we will continue to enable other OSs on ContainerUtils as we did until now. This is also why we decided to keep `systemsmanagement:Uyuni:Master:ContainerUtils` OBS project as a standalone project, instead of maintaining `uyuni-tools` directly in `systemsmanagement:Uyuni:Master`. But resources are not unlimited and we also need to prioritize the work to be done. The documentation specifies our "fully supported" OS and client tools and, as always, bug reports and in particular patches are more than welcomed. ;) 

So, in short, this doesn't mean that we are going to add and keep support for all the possible operating systems on the planet.  ;) 
The general idea in any case is that the collection of utilities under the umbrella of  `uyuni-tools` can be installed also on other systems, and from there you should be able to manage your Uyuni Server and Proxy and clients. 

When it comes to migrating from a legacy/classic RPM based Uyuni there are of course some potential constraints to keep in mind, but  when we will switch to "containers only", things should get easier.
